dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultAnalyticsService.java

```diff
@@ -1345,7 +1345,7 @@ private void addRawData( DataQueryParams params, Grid grid )
         List<DimensionalItemObject> items = Lists
                 .newArrayList( expressionService.getIndicatorDimensionalItemObjects( preprocessIndicators( indicators ) ) );
 
-        if ( items.isEmpty() )
+        if ( items.isEmpty() && params.getIndicators().isEmpty() )
```

Comment:
This check is _sort of_ pointless, in the sense that if there are no indicators in the query, we should not get here at all (i.e. break before the call to expression service), and if we get here, there should be indicators present, and then this check would always fail and hence could be removed (indicators would never be empty).

Reply:
The logic behind that check is the following: When we call the `expressionService` with indicators that contain ONLY static values (constants or numbers), the expression service will return `0` items, because it is not able to resolve the num/den of the indicator. So the condition `items.isEmpty()` is going to be `true` and we will exit and return no values.
So, doing this `params.getIndicators().isEmpty()` allows the system to proceed and calculate the static values.